### DISCUSSION TITLE
iTwin attempting to import styles during unit test

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI#1978] [Q&A] [answered]
After updating our app to iTwinUI-React v3, some components are attempting to import style sheets.

![image](https://github.com/iTwin/iTwinUI/assets/15036346/1dd6e1e4-aa1e-4037-afdc-1a7f8826bd7d)

We are wrapping the component in a `<ThemeProvider>`, which I believe should be enough to prevent it from attempting to fetch.

Specifically, this is happening when trying to click the Filter button in a table header. You can see a reproduction here:
https://stackblitz.com/edit/github-girbn1-bpujfc?file=src%2FApp.test.tsx

(Stop the app and run `npm run test`)

![image](https://github.com/iTwin/iTwinUI/assets/15036346/7a730c0f-bf45-40b3-a4eb-3580bea0d57d)

Adding `includeCss={false}` does not help. I'm curious if perhaps there is another ThemeProvider added around the Filter popover, which uses `theme="inherit"` and this is what is attempting to fetch the styles?

Any suggestions?

**Comment by mayank99** [comment]
This shouldn't be happening because we bail in test environments.

https://github.com/iTwin/iTwinUI/blob/3d42bc1e6eed059b3579b6898c6ccfe663c47fe1/packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx#L308-L309

https://github.com/iTwin/iTwinUI/blob/3d42bc1e6eed059b3579b6898c6ccfe663c47fe1/packages/itwinui-react/src/core/utils/functions/dev.ts#L6

Is the `jest` global not available in your testing setup?

I also could not get your repro link to run at all. 

```sh
 FAIL  src/App.test.tsx
  ● Test suite failed to run
    can't access lexical declaration 'App_1' before initialization
```

**Reply by HaveSpacesuit** [reply]
Hm, uncertain why you cannot run those tests on the stackblitz site...

It looks like globalThis.jest is undefined when I'm running jest tests. But even if I set it to a dummy object, it looks like it is still trying to fetch the styles.

**Reply by mayank99** [reply]
Okay I was able to make it run in Chrome but not Firefox. The problem appears to be that `jest` is set as a global, but `globalThis.jest` remains undefined. I think this is because jest does not support ESM.

So we just need to modify our check to use `typeof jest` instead of `typeof globalThis.jest`.

---

As your hunch here seems to be correct:

> I'm curious if perhaps there is another ThemeProvider added around the Filter popover

There is indeed a `ThemeProvider` used in our `Popover` component.

I also noticed that `Popover` uses `FloatingPortal` which portals the content (and the `ThemeProvider`) outside the application's main `ThemeProvider`. This may not cause a real problem in most cases though.

https://github.com/iTwin/iTwinUI/blob/3d42bc1e6eed059b3579b6898c6ccfe663c47fe1/packages/itwinui-react/src/core/Popover/Popover.tsx#L296-L301

**Reply by mayank99** [reply, marked as answer]
fixed in [3.9.0](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.9.0) 🚀
